src/global/BaseTest.java

```diff
@@ -39,12 +39,23 @@ public String getItemAtIndex(int index) {
         try {
             if (matcher.find()) return matcher.group(index);
             else fail("Your code's output did not follow the correct structure/syntax");
-        } catch (ArrayIndexOutOfBoundsException e) {
-            fail("The specified group doesn't exist");
+        } catch (IndexOutOfBoundsException e) {
+            fail("The specified group (" + (index) + ") doesn't exist");
         }
         return "";  // TODO: logically how does this behave?
     }
 
+    public String getItemByName(String name) {
+        Clause[] regSen = getRegexSentence();
+        int index = -1;
+        for (int i = 0; i < regSen.length; i++) {
```

Comment:
No need for the `index` variable I think. Just return immediately on line 53 with `i + 1` as the argument.

For error handling just manually throw the error on line 55